util.go

```diff
@@ -52,6 +52,10 @@ func copyEmptyStructToIfaces(in []struct{}) []interface{} {
 	return ifaces
 }
 
+func MultiaddrSplit(addr ma.Multiaddr) (peer.ID, ma.Multiaddr, error) {
```

Comment:
I shadowed multiaddrSplit with an exported version but probably shouldn't keep it this way.  I wanted to check if you think it is OK to export.